beacon_node/network/src/subnet_service/mod.rs

```diff
@@ -17,12 +32,645 @@ pub enum SubnetServiceMessage {
     Unsubscribe(Subnet),
     /// Add the `SubnetId` to the ENR bitfield.
     EnrAdd(Subnet),
-    /// Remove the `SubnetId` from the ENR bitfield.
-    EnrRemove(Subnet),
     /// Discover peers for a list of `SubnetDiscovery`.
     DiscoverPeers(Vec<SubnetDiscovery>),
 }
 
+use crate::metrics;
+
+/// The minimum number of slots ahead that we attempt to discover peers for a subscription. If the
+/// slot is less than this number, skip the peer discovery process.
+/// Subnet discovery query takes at most 30 secs, 2 slots take 24s.
+pub(crate) const MIN_PEER_DISCOVERY_SLOT_LOOK_AHEAD: u64 = 2;
+/// The fraction of a slot that we subscribe to a subnet before the required slot.
+///
+/// Currently a whole slot ahead.
+const ADVANCE_SUBSCRIBE_SLOT_FRACTION: u32 = 1;
+
+/// The number of slots after an aggregator duty where we remove the entry from
+/// `aggregate_validators_on_subnet` delay map.
+const UNSUBSCRIBE_AFTER_AGGREGATOR_DUTY: u32 = 2;
+
+/// A particular subnet at a given slot.
+#[derive(PartialEq, Eq, Hash, Clone, Debug, Copy)]
+pub struct ExactSubnet {
+    /// The `SubnetId` associated with this subnet.
+    pub subnet: Subnet,
+    /// For Attestations, this slot represents the start time at which we need to subscribe to the
+    /// slot. For SyncCommittee subnet id's this represents the end slot at which we no longer need
+    /// to subscribe to the subnet.
+    // NOTE: There was different logic between the two subscriptions and having a different
+    // interpretation of this variable seemed like the best way to group the logic, even though it
+    // may be counter-intuitive (apologies to future readers).
+    pub slot: Slot,
```

Comment:
I don't see where this field is used for `SyncCommittee`?

Reply:
Yeah good catch. I did multiple passes of this, to try and make the logic a bit easier to understand and ended up removing this type from sync committee logic. 

Have updated the comment.